### PR TITLE
Feature/#1199/add filtration and sorting for promo codes

### PR DESCRIPTION
**Pull request by anjtka333**
## Description
Filtration for promocodes added

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: helpers/filter-helper.js [hunk 1]
```diff
@@ -6,6 +6,102 @@ const {
...
+            name: searchPattern,
+          },
+          {
+            code: searchPattern,
+          },
+        ],
+      };
+    }
+
+    if (Object.keys(searchFilter).length) {
```
Comment (maarekua) [inline helpers/filter-helper.js:51]: Remove if.

File: helpers/filter-helper.js [hunk 2]
```diff
@@ -6,6 +6,102 @@ const {
 } = require('../consts/user-block-period');
 
 class FilterHelper {
+  filterByDateOrName(filter, search) {
```
Comment (maarekua) [inline helpers/filter-helper.js:9]: (filter = {}, search = '')

File: helpers/filter-helper.js [hunk 3]
```diff
@@ -6,6 +6,102 @@ const {
...
+    const regDate = /^\d+[.]\d+[.]\d+$/;
+
+    const searchTrimmed = (search ?? '').trim();
```
Comment (maarekua) [inline helpers/filter-helper.js:12]: search ?? - remove

File: helpers/filter-helper.js [hunk 4]
```diff
@@ -6,6 +6,102 @@ const {
 } = require('../consts/user-block-period');
 
 class FilterHelper {
+  filterByDateOrName(filter, search) {
+    const regDate = /^\d+[.]\d+[.]\d+$/;
```
Comment (maarekua) [inline helpers/filter-helper.js:10]: remove regexp

File: helpers/filter-helper.js [hunk 5]
```diff
@@ -6,6 +6,102 @@ const {
...
+
+    const searchTrimmed = (search ?? '').trim();
+
+    if (!searchTrimmed) {
+      return;
+    }
+    let searchFilter;
+
+    if (regDate.test(searchTrimmed)) {
```
Comment (maarekua) [inline helpers/filter-helper.js:19]: remove reg if

**Review by maarekua — CHANGES_REQUESTED** [review]

File: helpers/filter-helper.js [hunk 6]
```diff
@@ -6,6 +6,89 @@ const {
...
+          name: searchPattern,
+        },
+        {
+          code: searchPattern,
+        },
+      ],
+    };
+
+    if (!filter['$and']) {
+      filter['$and'] = [];
```
Comment (maarekua) [inline helpers/filter-helper.js:39]: filter['$and'] = [searchFilter];

**Review by maarekua — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Review by Yevhendzyuba — APPROVED** [review]

**Pull request merged**